src/hotspot/cpu/aarch64/nativeInst_aarch64.cpp

```diff
@@ -530,7 +530,7 @@ address NativeCall::trampoline_jump(CodeBuffer &cbuf, address dest) {
   MacroAssembler a(&cbuf);
   address stub = NULL;
 
-  if (a.far_branches()
+  if (a.codecache_branch_needs_far_jump()
```

Comment:
With this patch I do not change trampoline calls. I change far_jump and far_call procedures only.
Instead of far_branches() function we have two functions:
- codecache_branch_needs_far_jump to find if we need a far jump for intra-codecache branches
- codestub_branch_needs_far_jump to find if we need a far branch for codecache-to-nonmethodEntrypoint branch
So in this place I leave codecache_branch_needs_far_jump as exact equivalent of former far_branches() call.

Reply:
Ok. For the sake of formal logic here, let me introduce trampoline_needs_far_jump() function:
```
  if (a.trampoline_needs_far_jump() && ..) {
    stub = a.emit_trampoline_stub(dest);
  }
  if (stub == NULL) {
    // If we generated no stub, patch this call directly to dest.
    set_destination(dest);
```

Reply:
According to the comments of `trampoline_jump`:
```
// Generate a trampoline for a branch to dest.  If there's no need for a
// trampoline, simply patch the call directly to dest.
```
`trampoline_jump` is a jump or a far jump via a trampoline. It is not that a trampoline needs either a jump or a far jump.
What about:
```
if (a.jump_needs_trampoline() && ..) {
    stub = a.emit_trampoline_stub(dest);
}
```
Another variant:
```
if (a.is_trampoline_needed() && ..) {
    stub = a.emit_trampoline_stub(dest);
}
```